docs/_docs/04_community/client_factory.md

```diff
@@ -0,0 +1,176 @@
+# Azure SDK Client Factory
+
+This documentation provides and overview of the `client_factory.go` module, targeted use cases, and behaviors.  This module is intended to provide support for and simplify working with Azure's multiple cloud environments (Azure Public, Azure Government, Azure China, Azure Germany and Azure Stack).  Developers looking to contribute to additional support for Azure to TerraTest should leverage client_factory and use the patterns below to add a resource REST client from Azure Go SDK.  By doing so, it provides a consistent means for developers using TerraTest to test their Azure Infrastructure to connect to the correct cloud and its associated REST apis.
+
+## Background
+
+The Azure REST APIs support both Public and sovereign cloud environments (at the moment this includes Public, US Government, Germany, China, and Azure Stack environments).  If you are interacting with an environment other than public cloud, you need to set the base URI for the Azure REST API you are interacting with.
+
+### Base URI
+
+You must use the correct base URI's for the Azure REST API's (either directly or via Azure SDK for GO) to communicate with a cloud environment other than Azure Public. The Azure Go SDK supports this by using the `WithBaseURI` suffixed calls when creating service clients. For example, when using the `VirtualMachinesClient` with the public cloud, a developer would normally write code for the public cloud like so:
+
+```golang
```

Comment:
One of the benefits of being in the docs site is you can include snippets from real examples. You add them to https://github.com/gruntwork-io/terratest/blob/master/docs/_data/examples.yml and can then include them using an `include`. See https://github.com/gruntwork-io/terratest/blob/d0363f939d71a682726a72d2f82787e43e324b28/docs/_docs/01_getting-started/quick-start.md#example-1-terraform-hello-world for an example and https://terratest.gruntwork.io/docs/getting-started/quick-start/#example-1-terraform-hello-world for how it's rendered. 

This will make the code snippets more maintainable, as they'll actually go through our test suite.

Reply:
Leaving these 2 code samples as-is as they are intended to show an approach, not working code. Can revisit if needed.